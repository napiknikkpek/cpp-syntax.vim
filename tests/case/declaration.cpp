int a;
int & a;
int const & a;
const int a;
const int & a;

auto a;
auto & a;
auto const & a;
const auto & a;
auto && a;
auto & [a, b] = pair;
auto && [a, b] = pair;
const auto & [a, b] = pair;

Foo f;
Foo* f;
Foo *f;
Foo * f;
Foo const * const f;
Foo * const f;
const Foo f;
const Foo* f;
const Foo * const f;

Foo& f;
Foo &f;
Foo & f;
Foo const & f;

Foo&& f;
Foo &&f;
Foo && f;
Foo const && f;

Foo{};
Foo{1, 2};

const std::vector<int>::iterator it;

A<const T> a;

Foo foo{};
std::vector<int> foo{};

ranges::range auto X;

return Foo{};
