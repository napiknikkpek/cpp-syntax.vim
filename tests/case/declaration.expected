zzz.x.
zzz...x.
zzz.aaaaa...x.
aaaaa.zzz.x.
aaaaa.zzz...x.

aaaa.x.
aaaa...x.
aaaa.aaaaa...x.
aaaaa.aaaa...x.
aaaa....x.
aaaa....x..x....xxxx.
aaaa.....x..x....xxxx.
aaaaa.aaaa....x..x....xxxx.

ttt.x.
ttt..x.
ttt..x.
xxx...x.
ttt.aaaaa...aaaaa.x.
ttt...aaaaa...x.
aaaaa.ttt.x.
aaaaa.ttt..x.
aaaaa.ttt...aaaaa.x.

ttt..x.
ttt..x.
xxx...x.
ttt.aaaaa...x.
aaaaa.ttt...x.

ttt...x.
ttt...x.
xxx....x.
ttt.aaaaa....x.
aaaaa.ttt....x.

ttt...
ttt.c..c..

aaaaa.ttt..tttttt.zzz...tttttttt.xx.

t.aaaaa.t..x.

ttt.xxx...
ttt..tttttt.zzz..xxx...

aaaaaa.ttt...
